Marlin/src/gcode/feature/pause/M125.cpp

```diff
@@ -90,17 +100,44 @@ void GcodeSuite::M125() {
 
   const bool sd_printing = TERN0(SDSUPPORT, IS_SD_PRINTING());
 
-  ui.pause_show_message(PAUSE_MESSAGE_PARKING, PAUSE_MODE_PAUSE_PRINT);
+  //Maintenance park
+  if (parser.seen('T')) {
+    if (ENABLED(HAS_MARLINUI_MENU)){
+      show_m125t(1);
+      maintenance_park_enabled = true;
+      const bool show_lcd = false;
+      if (pause_print(retract, park_point, show_lcd, 0)) {
+        millis_t start_time = millis();
+        bool flipflap = 0;
+        uint8_t can_beep = 0;
+        while(maintenance_park_enabled){
+          idle();
+          if ( (millis() - start_time) > 1000 ) {
+            start_time = millis();
+            if (parser.seenval('T') && can_beep == parser.linearval('T')) { BUZZ(100, 1000); can_beep = 0; }
+            show_m125t(flipflap);
+            flipflap = flipflap? 0:1;
+            can_beep++;
+          }
+        }
+      }
+      resume_print(0, 0, -retract, 0);
+    }
+  }
+  else {
+    ui.pause_show_message(PAUSE_MESSAGE_PARKING, PAUSE_MODE_PAUSE_PRINT);
 
-  // If possible, show an LCD prompt with the 'P' flag
-  const bool show_lcd = TERN0(HAS_MARLINUI_MENU, parser.boolval('P'));
+    // If possible, show an LCD prompt with the 'P' flag
+    const bool show_lcd = TERN0(HAS_MARLINUI_MENU, parser.boolval('P'));
 
-  if (pause_print(retract, park_point, show_lcd, 0)) {
-    if (ENABLED(EXTENSIBLE_UI) || BOTH(EMERGENCY_PARSER, HOST_PROMPT_SUPPORT) || !sd_printing || show_lcd) {
-      wait_for_confirmation(false, 0);
-      resume_print(0, 0, -retract, 0);
+    if (pause_print(retract, park_point, show_lcd, 0)) {
+      if (ENABLED(EXTENSIBLE_UI) || BOTH(EMERGENCY_PARSER, HOST_PROMPT_SUPPORT) || !sd_printing || show_lcd) {
+        wait_for_confirmation(false, 0);
+        resume_print(0, 0, -retract, 0);
```

Comment:
if resume is inside this if scope and if, for some reason, is not true (my case with reprap full discount lcd) you don't call resume_print and print (when using SD) will hang. This is what I fixed in my PR #24877

Reply:
Thks for the fix , my PR, didn't make modification of the noticed lines , my code makes a timeless pause just before. I will rebase it when your fix will be on the upstream. I have MKS 12864 screen and I don't have this bug.

Reply:
Question: sd_printing means ' SD printing in progress ' or ' SD printing in progress but in pause ' ,  I'm not sur ' sd_printing is pause flag'

Reply:
My cnf is emergency parser and no host prompt. And since reprap full discount is not EXTENSIBLE_UI and I was printing via SD then printer was not "waiting for confirmation" but more dangerous with no "resume_print" call it was stuck waiting to resume. Just moving resume out of scope it solve printer hanging